src/Saunter/AsyncApiEndpointRouteBuilderExtensions.cs

```diff
@@ -17,12 +17,29 @@ public static class AsyncApiEndpointRouteBuilderExtensions
                 .Build();
 
             // Retrieve the route from options. If options can't be retrieved, use default constant
-            var options= endpoints.ServiceProvider.GetService<IOptions<AsyncApiOptions>>();
+            var options = endpoints.ServiceProvider.GetService<IOptions<AsyncApiOptions>>();
             var route = options?.Value?.Middleware?.Route ?? AsyncApiMiddlewareOptions.AsyncApiMiddlewareDefaultRoute;
 
             // Add the endpoint
             return endpoints.Map(route, pipeline).WithDisplayName("Async API Documentation");
         }
+
+        public static IEndpointConventionBuilder MapAsyncApiUi(
+            this IEndpointRouteBuilder endpoints)
+        {
+            // Add the middleware 
+            var pipeline = endpoints.CreateApplicationBuilder()
+                .UseMiddleware<AsyncApiUiMiddleware>()
+                .Build();
+
+            // Retrieve the route from options. If options can't be retrieved, use default constant
+            var options = endpoints.ServiceProvider.GetService<IOptions<AsyncApiOptions>>();
+            var route = "/asyncapi/ui/{resource}";
```

Comment:
If you register a route like 
```csharp
var route = "/asyncapi/ui/{*resource}";
```
it is a catch-all for `/asyncapi/ui/any/number/of/segments`.

Which prevents you from having to register the 
```csharp
var subRoute = "/asyncapi/ui/{resource}/{subresource}";
```
route too.

Reply:
I'll make that update.